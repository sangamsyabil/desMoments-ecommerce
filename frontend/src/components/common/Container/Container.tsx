import { Box, makeStyles } from '@material-ui/core';
import clsx from 'clsx';
import React from 'react';
import { CustomTheme } from '../../../theme/muiTheme';
import { Text } from '../Text/Text';

export interface ContainerProps {
  fontWeight?: 'bold' | 'normal';
  textTransform?: 'uppercase' | 'lowercase' | 'capitalize';
  externalStyle?: string;
  variant?:
    | 'button'
    | 'caption'
    | 'h1'
    | 'h2'
    | 'h3'
    | 'h4'
    | 'h5'
    | 'h6'
    | 'inherit'
    | 'subtitle1'
    | 'subtitle2'
    | 'body1'
    | 'body2'
    | 'overline'
    | 'srOnly'
    | undefined;
  color?: 'inherit' | 'initial' | 'primary' | 'secondary' | 'textPrimary' | 'textSecondary' | 'error' | undefined;
}

const useStyles = makeStyles((theme: CustomTheme) => ({
  container: {
    borderColor: theme.palette.primary.main,
    borderWidth: 1.5,
    borderStyle: 'solid',
    textAlign: 'center',
    padding: theme.spacing(0.7),
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10,
    width: 'fit-content',
    cursor: 'pointer',
    '&:hover': {
      backgroundColor: theme.palette.primary.main,
      color: theme.palette.secondary.light,
    },
  },
}));

const Container: React.FC<ContainerProps> = ({
  children,
  textTransform,
  fontWeight,
  externalStyle,
  color,
  variant,
}) => {
  const classes = useStyles();
  return (
    <Box className={clsx([classes.container, externalStyle])}>
      <Text color={color} variant={variant} textTransform={textTransform} fontWeight={fontWeight}>
        {children}
      </Text>
    </Box>
  );
};

export { Container };
