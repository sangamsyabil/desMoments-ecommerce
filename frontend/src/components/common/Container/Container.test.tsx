import { render } from '@testing-library/react';
import React from 'react';
import { Container, ContainerProps } from './Container';

const mockData: ContainerProps = {
  variant: 'button',
  color: 'initial',
  textTransform: 'uppercase',
  fontWeight: 'bold',
};

test('Button renders correctly', () => {
  const { asFragment } = render(<Container {...mockData}>Mock button</Container>);
  expect(asFragment()).toMatchSnapshot();
});
