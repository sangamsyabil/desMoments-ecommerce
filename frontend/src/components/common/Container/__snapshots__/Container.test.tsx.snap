// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-2 makeStyles-container-1"
  >
    <span
      class="MuiTypography-root makeStyles-text-3 makeStyles-text-4 MuiTypography-button"
    >
      Mock button
    </span>
  </div>
</DocumentFragment>
`;
